From: Pino Toscano <pino@kde.org>
Date: Sat, 26 Nov 2016 09:26:16 +0100
Subject: java: set names in permission dialog

This makes sure the reply from the user is used and sent back.

Fixes commit b68911a60306e53123430b55f24a5fa5ab4e7873.

(manual backport of commit 856a96852f566ed7eb855810e347f95003c25017 and
commit bd4be07823e422fa15bf4273e0bb9df35b866e35 from khtml)
---
 khtml/java/kjavaappletserver.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/khtml/java/kjavaappletserver.cpp b/khtml/java/kjavaappletserver.cpp
index cece008aa3..7246db9f54 100644
--- a/khtml/java/kjavaappletserver.cpp
+++ b/khtml/java/kjavaappletserver.cpp
@@ -808,16 +808,20 @@ QString PermissionDialog::exec( const QString & cert, const QString & perm ) {
     buttonLayout->setObjectName("buttonLayout");
 
     QPushButton* const no = new QPushButton( i18n("&No"), dialog );
+    no->setObjectName( "no" );
     no->setDefault( true );
     buttonLayout->addWidget( no );
 
     QPushButton* const reject = new QPushButton( i18n("&Reject All"), dialog );
+    reject->setObjectName( "reject" );
     buttonLayout->addWidget( reject );
 
     QPushButton* const yes = new QPushButton( i18n("&Yes"), dialog );
+    yes->setObjectName( "yes" );
     buttonLayout->addWidget( yes );
 
     QPushButton* const grant = new QPushButton( i18n("&Grant All"), dialog );
+    grant->setObjectName( "grant" );
     buttonLayout->addWidget( grant );
     dialogLayout->addLayout( buttonLayout );
     dialog->resize( dialog->minimumSizeHint() );
